### ISSUE TITLE
force:source:deploy:  path must not be empty

### ISSUE THREAD

**Issue by steals** [forcedotcom/cli#684]
### Summary

I have a project with multiple package directories, i.e.
```
"packageDirectories": [
        {
            "path": "force-app",
            "default": false
        },
        {
            "path": "unpackaged",
            "default": true
        }
    ],
```
I also have a manifest file in a separate directory that is used by CI when deploying to another org.
when CI runs `sfdx force:source:deploy --manifest ./manifest/package.xml --targetusername CI_ORG` command to deploy it gets the following error:
`ERROR running force:source:deploy:  path must not be empty`

This command worked fine a few weeks ago, but today I've updated SFDX CLI to the latest version and it's started failing.
### Expected result

sfdx force:source:deploy command should succeed

### Actual result

Error `ERROR running force:source:deploy:  path must not be empty`

### Additional information

**SFDX CLI Version**: sfdx-cli/7.76.1-f301d5641f linux-x64 node-v12.18.3

**SFDX plugin Version**(to find the version of the CLI plugin run sfdx plugins --core):
@oclif/plugin-autocomplete 0.1.5 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.1 (core)
@oclif/plugin-update 1.3.10 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-trust 3.4.3 (core)
alias 1.1.2 (core)
analytics 1.12.1 (core)
config 1.1.9 (core)
generator 1.1.3 (core)
salesforcedx 50.1.1
├─ salesforce-alm 50.1.1
├─ templates 50.1.0
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7
├─ custom-metadata 1.0.10
└─ apex 0.1.1
sfdx-cli 7.76.1 (core)

**OS and version**: Linux, MacOS

**Comment by bobalicious** [comment]
Getting a similar result with 7.76.1 when running `force:mdapi:convert`

Not reproducible in 7.74.1.

Would add more detail, but have a lot to get done now...

**Comment by steals** [comment]
Reverting `salesforcedx` plugin to 49.13.1 might be a temporary workaround

**Comment by shetzel** [comment]
I'll try to repro.  If you can get a stack trace by appending `--dev-debug` and maybe a few details about the contents of the package.xml (does it contain components from both packages?) that would help expedite.  Thank you!

**Comment by steals** [comment]
Here is some part from the middle in the debug output:
```
sfdx:SfdxProjectJson WARN The config file: /Users/steals/.sfdx/sfdx-project.json is not schema valid
sfdx:SfdxProjectJson Due to: Validation errors:
sfdx:SfdxProjectJson should have required property 'packageDirectories' +4ms
sfdx:SfdxProjectJson INFO Reading config file: /Users/steals/Workspaces/project_name/sfdx-project.json +1ms
sfdx:SfdxProjectJson TRACE Setup child 'SchemaValidator' logger instance +6ms
sfdx:SchemaValidator DEBUG Schema loaded for /Users/steals/.local/share/sfdx/node_modules/@salesforce/schemas/sfdx-project.schema.json +11ms
ERROR running force:source:push:  path must not be empty 
*** Internal Diagnostic ***

TypeError: path must not be empty
  at throwError (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:366:9)
  at checkPath (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:379:12)
  at Ignore._test (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:502:5)
  at Ignore.ignores (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:541:17)
  at ForceIgnore.denies (/Users/steals/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/src/metadata-registry/forceIgnore.ts:55:28)
  at SourcePathStatusManager.getSourcePathInfos (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePathStatusManager.js:206:34)
  at SourceWorkspaceAdapter.init (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourceWorkspaceAdapter.js:76:31)
  at Function.create (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/kit/lib/creatable.js:31:9)
  at MdapiPushApi.init (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePushApi.js:32:20)
  at Function.create (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/kit/lib/creatable.js:31:9)
  at SourceApiCommand.execute (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourceApiCommand.js:48:15)
  at SourcePushCommand.execLegacyCommand (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/ToolbeltCommand.js:149:29)
  at SourcePushCommand.run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/commands/force/source/push.js:30:16)
  at SourcePushCommand._run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/command/lib/sfdxCommand.js:93:40)
  at Config.runCommand (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/config/lib/config.js:173:24)
  at Main.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/main.js:27:9)
Outer stack:
  at Function.wrap (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/core/lib/sfdxError.js:171:27)
  at SourcePushCommand.catch (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/ToolbeltCommand.js:248:46)
  at SourcePushCommand._run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/command/lib/sfdxCommand.js:97:13)
  at Config.runCommand (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/config/lib/config.js:173:24)
  at Main.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/main.js:27:9)
  at Main._run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/command.js:43:20)
  at Object.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/dist/cli.js:32:20)
******

sfdx:SourcePushCommand ERROR [
sfdx:SourcePushCommand   '\u001b[1mERROR running force:source:push: \u001b[22m',
sfdx:SourcePushCommand   '\u001b[31mpath must not be empty\u001b[39m',
sfdx:SourcePushCommand   '\u001b[31m\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m*** Internal Diagnostic ***\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31mTypeError: path must not be empty\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at throwError (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:366:9)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at checkPath (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:379:12)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Ignore._test (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:502:5)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Ignore.ignores (/Users/steals/.local/share/sfdx/node_modules/ignore/index.js:541:17)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at ForceIgnore.denies (/Users/steals/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/src/metadata-registry/forceIgnore.ts:55:28)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePathStatusManager.getSourcePathInfos (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePathStatusManager.js:206:34)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourceWorkspaceAdapter.init (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourceWorkspaceAdapter.js:76:31)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Function.create (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/kit/lib/creatable.js:31:9)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at MdapiPushApi.init (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePushApi.js:32:20)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Function.create (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/kit/lib/creatable.js:31:9)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourceApiCommand.execute (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourceApiCommand.js:48:15)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePushCommand.execLegacyCommand (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/ToolbeltCommand.js:149:29)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePushCommand.run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/commands/force/source/push.js:30:16)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePushCommand._run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/command/lib/sfdxCommand.js:93:40)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Config.runCommand (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/config/lib/config.js:173:24)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Main.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/main.js:27:9)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31mOuter stack:\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Function.wrap (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/core/lib/sfdxError.js:171:27)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePushCommand.catch (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/ToolbeltCommand.js:248:46)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at SourcePushCommand._run (/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/command/lib/sfdxCommand.js:97:13)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Config.runCommand (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/config/lib/config.js:173:24)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Main.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/main.js:27:9)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Main._run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/node_modules/@oclif/command/lib/command.js:43:20)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m    at Object.run (/Users/steals/.local/share/sfdx/client/7.76.1-f301d5641f/dist/cli.js:32:20)\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m******\u001b[39m\n' +
sfdx:SourcePushCommand     '\u001b[31m\u001b[39m'
sfdx:SourcePushCommand ] +2s
sfdx:core TRACE Setup child 'Org' logger instance +2s
sfdx:core TRACE Setup child 'Config' logger instance +1ms
@oclif/config start postrun hook +2s
```

There is nothing special in the manifest. Mostly it contains wildcards for common types like AuraComponents, ApexClasses, etc.

**Comment by celliott181** [comment]
Just ran into this issue after upgrading from `7.75.1-5585088c75` to `7.76.1-f301d5641f`. Previously working scratch orgs and newly generated scratch orgs exhibit this behavior.

It also happens for `sfdx force:source:push` commands.

**Comment by shetzel** [comment]
@steals - thanks for the stack trace.  Would you mind updating this file:
`/Users/steals/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePathStatusManager.js`
Just before line 206 that has the condition:
`if (this.forceIgnore.denies(sourcePathInfo.sourcePath)) {`
to do this:
`if (!sourcePathInfo.sourcePath) console.dir(sourcePathInfo)`

Please let me know what the output of the console.dir() is.

That code is checking if the metadata should be forceignored but the file path is empty somehow.  It would be interesting to see what the entry looks like in sourcePathInfos.json and how the sourcePath became empty.

**Comment by steals** [comment]
@shetzel `console.dir` doesn't print anything. If I remove if statement, then I can see similar logs for every metadata file
```
SourcePathInfo {
  sourcePath: '/Users/steals/Workspaces/project_name/unpackaged/pages/Sample.page-meta.xml',
  deferContentHash: false,
  isWorkspace: false,
  isArtifactRoot: false,
  state: 'u',
  package: 'unpackaged',
  isDirectory: false,
  isMetadataFile: true,
  metadataType: 'ApexPage',
  size: 443,
  modifiedTime: 1595429951549,
  changeTime: 1595429951549,
  contentHash: 'b891e687b878da2e0fe5c498e69eec43db82d824'
}
```

**Comment by steals** [comment]
Ok, I've wrapper line 206 into try catch block, i.e.
```
try {
    if (this.forceIgnore.denies(sourcePathInfo.sourcePath)) {
            shouldIncludeSourcePathInfo = false;
    }
} catch (er) {
    console.log('*** exception');
    console.dir(sourcePathInfo);
}
```
 and get the following log entry:
```
*** exception
SourcePathInfo {
  sourcePath: '/Users/steals/Workspaces/project-name',
  deferContentHash: false,
  isWorkspace: true,
  isArtifactRoot: false,
  state: 'u',
  isDirectory: true,
  isMetadataFile: false,
  size: 960,
  modifiedTime: 1603117017126,
  changeTime: 1603117017126,
  contentHash: '9dba9739ba2a966624714ce7ed3f04c4cf5bdbfd'
}
```
where `project-name` is a root folder of the project.

**Comment by steals** [comment]
Here are the steps to reproduce with a new project:

- sfdx force:project:create --projectname tempproj
- cd tempproj
- sfdx force:org:create -f config/project-scratch-def.json --setalias tempproj --durationdays 1 --setdefaultusername
- sfdx force:source:push

**Comment by shanebird92** [comment]
I ran into this exact same problem this morning. 
Seems it was related to my force ignore file so I deleted it and the retrieve and deploy's work now.
I haven't jumped back in to check exactly the issue in the force-ignore but it's worth playing around with

Below was the debug output of my retrieve command

ERROR running force:source:retrieve:  path must not be empty 
*** Internal Diagnostic ***

TypeError: path must not be empty
    at throwError (/Users/shanebird/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/node_modules/ignore/index.js:366:9)
    at checkPath (/Users/shanebird/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/node_modules/ignore/index.js:379:12)
    at Ignore._test (/Users/shanebird/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/node_modules/ignore/index.js:502:5)
    at Ignore.ignores (/Users/shanebird/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/node_modules/ignore/index.js:541:17)
    at ForceIgnore.denies (/Users/shanebird/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/lib/src/metadata-registry/forceIgnore.js:49:34)
    at SourcePathStatusManager.getSourcePathInfos (/Users/shanebird/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/source/sourcePathStatusManager.js:206:34)
Outer stack:
    at Function.wrap (/Users/shanebird/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/core/lib/sfdxError.js:171:27)
    at SourceRetrieveCommand.catch (/Users/shanebird/.local/share/sfdx/node_modules/salesforce-alm/dist/ToolbeltCommand.js:248:46)
******

**Comment by danic85** [comment]
`sfdx plugins:install salesforcedx@49.13.1` resolved this for me (as a workaround at least)

**Comment by shetzel** [comment]
Thanks everyone!  We'll get a fix in asap.

Workarounds:
1. Downgrade by installing a previous version of the salesforcedx plugin
2. Wrap the code as mentioned by @steals above

Deleting the `.forceignore` file works too since that will prevent the check entirely, but if you want to ignore certain files or directories you should choose another workaround.

**Comment by yippie** [comment]
I also had to downgrade due to this

**Comment by ryanfollmer** [comment]
I was unable to use the downgrade because of this previous bug: https://github.com/forcedotcom/cli/issues/674. So, the alternate workaround -> wrapping the code as mentioned appears to work for me.

**Comment by shetzel** [comment]
Can those on this thread respond with the values of `version` and `dependencies.@salesforce/source-deploy-retrieve` from this file: `/Users/<username>/.local/share/sfdx/node_modules/salesforce-alm/package.json`
And the value of `version` from this file: `/Users/<username>/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/package.json`

Those are the file locations for a Mac installer default installation.  They could be other places if you changed the location or installed via npm.

Also verify a CLI version of 7.76.1 or greater and salesforcedx plugin version of 50.1.1 or greater.

**Comment by steals** [comment]
SFDX CLI sfdx-cli/7.77.0-112dd7d412 darwin-x64 node-v12.18.3
salesforcedx 50.2.0
├─ salesforce-alm 50.2.0
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7
├─ custom-metadata 1.0.10
├─ apex 0.1.1
└─ templates 50.1.0
sfdx-cli 7.77.0 (core)

/Users/<username>/.local/share/sfdx/node_modules/salesforce-alm/package.json:
```
"name": "salesforce-alm",
"version": "50.2.0",
"dependencies": {
    "@oclif/config": "^1.14.0",
    "@oclif/errors": "^1.2.2",
    "@oclif/plugin-help": "^2.2.3",
    "@salesforce/command": "^3.0.1",
    "@salesforce/core": "^2.13.0",
    "@salesforce/kit": "^1.3.2",
    "@salesforce/plugin-analytics": "^1.12.0",
    "@salesforce/source-deploy-retrieve": "^1.0.18",
    "@salesforce/ts-types": "~1.2.1",
    "adm-zip": "^0.4.13",
    "ansi-styles": "^3.2.1",
    "antlr4": "^4.7.2",
    "archiver": "^1.0.1",
    "bluebird": "^3.5.5",
    "bunyan-sfdx-no-dtrace": "^1.8.2",
    "chalk": "^2.4.2",
    "cli-ux": "^5.4.4",
    "csv-parse": "^4.6.5",
    "debug": "^3.2.6",
    "fast-xml-parser": "^3.17.4",
    "fs-extra": "^4.0.3",
    "gitignore-parser": "^0.0.2",
    "glob": "^7.1.6",
    "heroku-cli-util": "^8.0.12",
    "js2xmlparser": "^3.0.0",
    "jsen": "^0.6.6",
    "jsforce": "^1.9.3",
    "jsonwebtoken": "^8.5.1",
    "klaw": "^2.1.1",
    "lodash": "^4.17.19",
    "mime": "^1.6.0",
    "mkdirp": "^0.5.1",
    "moment": "^2.24.0",
    "open": "^7.2.0",
    "optional-js": "^1.3.1",
    "readline-sync": "^1.4.9",
    "replace": "^1.1.0",
    "request": "^2.88.0",
    "request-promise-native": "^1.0.5",
    "sfdx-faye": "^1.0.9",
    "strip-ansi": "^5.2.0",
    "xml2js": "0.4.19",
    "xmldom-sfdx-encoding": "^0.1.30"
  },
```

/Users/<username>/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/package.json
```
"name": "@salesforce/source-deploy-retrieve",
"version": "1.0.19",
```
Versions don't match.

**Comment by douglascayers** [comment]
### sfdx version
```
>sfdx --version
sfdx-cli/7.76.1-f301d5641f darwin-x64 node-v12.18.3
```

### sfdx plugins
```
>sfdx plugins --core
@oclif/plugin-autocomplete 0.2.0
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.1 (core)
@oclif/plugin-update 1.3.10 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-diff 0.0.6
@salesforce/sfdx-trust 3.4.3 (core)
alias 1.1.2 (core)
analytics 1.12.1 (core)
config 1.1.9 (core)
generator 1.1.3 (core)
salesforcedx 50.1.1
├─ templates 50.1.0
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7
├─ custom-metadata 1.0.10
├─ apex 0.1.1
└─ salesforce-alm 50.1.1
sfdx-cli 7.76.1 (core)
```

### /Users/doug.ayers/.local/share/sfdx/node_modules/salesforce-alm/package.json
```json
{
  "name": "salesforce-alm",
  "version": "50.1.1",
  ...
  "dependencies": {
    "@oclif/config": "^1.14.0",
    "@oclif/errors": "^1.2.2",
    "@oclif/plugin-help": "^2.2.3",
    "@salesforce/command": "^3.0.1",
    "@salesforce/core": "^2.13.0",
    "@salesforce/kit": "^1.3.2",
    "@salesforce/plugin-analytics": "^1.12.0",
    "@salesforce/source-deploy-retrieve": "^1.0.18",
    "@salesforce/ts-types": "~1.2.1",
    "adm-zip": "^0.4.13",
    "ansi-styles": "^3.2.1",
    "antlr4": "^4.7.2",
    "archiver": "^1.0.1",
    "bluebird": "^3.5.5",
    "bunyan-sfdx-no-dtrace": "^1.8.2",
    "chalk": "^2.4.2",
    "cli-ux": "^5.4.4",
    "csv-parse": "^4.6.5",
    "debug": "^3.2.6",
    "fast-xml-parser": "^3.17.4",
    "fs-extra": "^4.0.3",
    "gitignore-parser": "^0.0.2",
    "glob": "^7.1.6",
    "heroku-cli-util": "^8.0.12",
    "js2xmlparser": "^3.0.0",
    "jsen": "^0.6.6",
    "jsforce": "^1.9.3",
    "jsonwebtoken": "^8.5.1",
    "klaw": "^2.1.1",
    "lodash": "^4.17.19",
    "mime": "^1.6.0",
    "mkdirp": "^0.5.1",
    "moment": "^2.24.0",
    "open": "^7.2.0",
    "optional-js": "^1.3.1",
    "readline-sync": "^1.4.9",
    "replace": "^1.1.0",
    "request": "^2.88.0",
    "request-promise-native": "^1.0.5",
    "sfdx-faye": "^1.0.9",
    "strip-ansi": "^5.2.0",
    "xml2js": "0.4.19",
    "xmldom-sfdx-encoding": "^0.1.30"
  },
  ...
}
```

### /Users/doug.ayers/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/package.json
```json
{
  "name": "@salesforce/source-deploy-retrieve",
  "version": "1.0.19",
  "gusBuild": "49.1.0",
  ...
  "dependencies": {
    "@salesforce/core": "2.9.0",
    "archiver": "4.0.1",
    "fast-xml-parser": "^3.17.4",
    "ignore": "^5.1.8",
    "xmldom-sfdx-encoding": "^0.1.29"
  },
  ...
}
```

**Comment by yippie** [comment]
**~/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/package.json** 
version: 1.1.7

**~/.local/share/sfdx/node_modules/salesforce-alm/package.json**
  "version": "47.15.0",
dependencies.@salesforce/source-deploy-retrieve NOT LISTED

 **sfdx plugins --core**
@mavens/mavens-cli 1.0.0-6 (link) /Users/kaiamundsen/.local/share/sfdx/node_modules/@mavens/mavens-cli
@mavens/mic-cli-plugin 9.0.0-0 (link) /Users/kaiamundsen/workspace/MedInfoCloud-CLI-Plugin
@oclif/plugin-autocomplete 0.1.5 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.1 (core)
@oclif/plugin-update 1.3.10 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-scanner 2.3.0
@salesforce/sfdx-trust 3.4.3 (core)
alias 1.1.2 (core)
analytics 1.12.1 (core)
config 1.1.9 (core)
generator 1.1.3 (core)
salesforcedx 50.2.0
├─ salesforce-alm 50.2.0
├─ templates 50.1.0
├─ custom-metadata 1.0.10
├─ apex 0.1.1
└─ @salesforce/sfdx-plugin-lwc-test 0.1.7
sfdmu 3.7.17
sfdx-cli 7.76.1 (core)
texei-sfdx-plugin 1.9.0

**sfdx --version**
sfdx-cli/7.76.1-f301d5641f darwin-x64 node-v12.18.3

**Comment by ChristopherMGlass** [comment]
Working with @yippie I also ran into this issue
**sfdx version**
sfdx-cli/7.77.0-112dd7d412 darwin-x64 node-v12.18.3

**~/.local/share/sfdx/node_modules/@salesforce/source-deploy-retrieve/package.json**
version: 1.1.3

**~/.local/share/sfdx/node_modules/salesforce-alm/package.json**
"version": "47.15.0",
dependencies.@salesforce/source-deploy-retrieve NOT LISTED

salesforcedx 50.2.0

**Comment by shetzel** [comment]
Thanks everyone.  Those of you who listed a source-deploy-retrieve library version of 1.0.19 in node_modules will definitely see the issue.  That library doesn't have the changes necessary to work properly in salesforce-alm plugin v50.2.0.  It satisfies the caret version defined in that plugin's package.json (^1.0.18), and that is a bug.  We will be updating to require the proper minimum version of that library as well as making some changes in the library.  We hope to update the latest-rc tag with those changes sometime today.

@yippie @ChristopherMGlass 
v1.1.7 of source-deploy-retrieve has the necessary code to work with salesforce-alm v50.2.0
v1.1.3 of source-deploy-retrieve does not have the necessary code to work with salesforce-alm v50.2.0

It looks like both of you have installed salesforcedx specifically to version 50.2.0 so the files to check are probably in a different location.  If you `sfdx plugins:uninstall salesforcedx` you'll go back to the CLI bundled version for v7.76.1 and then the file should have a dependency defined for the source-deploy-retrieve library.  v47.15.0 won't have that dependency.

Sorry for all the disruption this has caused!  We strive to constantly improve experience with the CLI and this was a miss.  Thanks to everyone for providing details to help us fix things.

**Comment by steals** [comment]
Looks like upgrading `salesforcedx` plugin to v50.3.1 resolved the issue. I've run `sfdx plugins:install salesforcedx@50.3.1` to  upgrade

**Comment by shetzel** [comment]
An update to salesforcedx latest-rc made today should hopefully fix these issues.  This will become the default core version shipped with the CLI release planned for this Thursday.
You can try the fixes made in the latest-rc by:
`sfdx plugins:install salesforcedx@latest-rc`

This is the same version @steals mentioned (50.3.1).  The latest-rc label now points to that version.

**Comment by MayTheSForceBeWithYou** [comment]
Thank you @shetzel ! My dx push was broken ("path must not be empty"), borking our python scratch org setup script.  This fixed it!  I can now get a cup of coffee while my script is running worry-free.

**Comment by uvaiskomathsfdc** [comment]
Had the same issue and i uninstalled DX.

`sfdx plugins:uninstall salesforcedx`

Then installed default v50.3.1

`sfdx plugins:install salesforcedx`

Solved my issue

**Comment by shetzel** [comment]
This is fixed as of: salesforcedx v50.5.0 (November 12, 2020) - CLI 7.80.0
